src/endpoint/s3/s3_rest.js

```diff
@@ -210,10 +210,14 @@ async function authorize_request_policy(req) {
     const auth_token = req.object_sdk.get_auth_token();
     const arn_path = _get_arn_from_req_path(req);
     const method = _get_method_from_req(req);
+    const condition_elements = {};
+    condition_elements.encryption = s3_utils.parse_encryption(req);
+    condition_elements.bucket = req.params.bucket;
+    condition_elements.tagging = await req.object_sdk.get_object_tagging(req.params)?.tagging;
```

Comment:
why do we always get the tagging here? this is always needed? can't we get it on demand?